tests/network/samba/samba_adcli.pm

```diff
@@ -0,0 +1,166 @@
+
+    foreach my $service (qw(smb nmb winbind sssd)) {
+        systemctl("enable --now $service");
+    }
+    systemctl('restart nscd');
+}
+
+sub update_password {
+    # Invalidate the password of the local computer account on AD
+    script_retry("adcli update --verbose --computer-password-lifetime=0 --domain '$AD_domain'", retry => 3, delay => 60, fail_message => "Error invalidating local password");
+    # Restore the password with --add-samba-data as requested by poo#91950
+    script_retry("adcli update --verbose --computer-password-lifetime=0 --domain '$AD_domain' --add-samba-data", retry => 3, delay => 60, fail_message => "Error re-adding password with samba data");
+
+    # Check the trust secret for the domain
+    if (script_run("wbinfo -tP") != 0) {
+        my $output = script_output('wbinfo -tP', proceed_on_failure => 1);
+
+        # Check for bsc#1188575
+        if ($output =~ "WBC_ERR_AUTH_ERROR") {
+            die("wbinfo output failed") unless (is_sle('=12-SP3') || is_sle('=12-SP4'));
+            record_soft_failure("bsc#1188575");
+        }
+    }
+}
+
+sub randomize_hostname {
+    my $hostname = "openqa-" . random_string(length => 8);
+    assert_script_run("hostnamectl set-hostname '$hostname'");
+}
+
+sub run {
+    select_serial_terminal;
+
+    # Ensure the required variables are set
+    my $password = get_required_var("_SECRET_AD_DOMAIN_PASSWORD");
+    define_secret_variable("AD_DOMAIN_PASSWORD", $password);
+
+    samba_sssd_install();
+    randomize_hostname();    # Prevent race condition with parallel test runs
+    join_domain();
+    $domain_joined = 1;
+
+    # Verify users and groups from AD via winbind.
+    # Note: The following checks are subject to sporadic failures (poo#96513)
+    record_soft_failure("poo#96513 - Failed to get AD domain") if (script_run("wbinfo -D $AD_domain", timeout => 120) != 0);
+    record_soft_failure("poo#96513 - Failed to get AD username") if (script_run("wbinfo -u | grep 'geekotest'", timeout => 120) != 0);
+    record_soft_failure("poo#96513 - Failed to get AD groups") if (script_run("wbinfo -g | grep 'openqa'", timeout => 120) != 0);
+    record_soft_failure("poo#96513 Failed to get AD user info for geekotest") if (script_run("wbinfo -i geekotest\@$AD_domain", timeout => 120) != 0);
+
+    # poo#91950 (update password with adcli --add-samba-data option)
+    update_password() unless (is_sle("=15") || is_sle("<12-SP4"));    # sle 15 and 12-SP3 do not support the `--add-samba-data` option
+
+    assert_script_run("echo \"\$AD_DOMAIN_PASSWORD\" | net ads leave --domain '$AD_hostname' -U Administrator -i", fail_message => "Failed to leave the domain (poo#96986)");
+    $domain_joined = 0;
+
+    # For futher extensions
+    # - Mount //GEEKO
+    # - smbclient //10.162.30.119/openQA as geekouser will be denied, as berhard is the owner
+    # - delete the computer OU after the test is done in post_run_hook
+    # - test winbind (samba?) authentication
```

Comment:
Can you please update these lines if the AD configuration is controlled via variables?

Reply:
Yep, done.